packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,349 @@
+                do {
+                    input = await CliUtils.readPrompt(TextUtils.chalk.green(`The remote version of the document you were editing has changed. `+
+                    `View the updated version of the mainframe file before proceeding with edits? y/n`));
+                }
+                while (input === '');
+                if (input === null) {
+                    throw new ImperativeError({
+                        msg: TextUtils.chalk.red(`No input provided. Command terminated. Stashed file will persist: ${tempPath}`)
+                    });
+                }
+                return input.toLowerCase() === 'y';
+        }
+    }
+
+    /**
+     * Download file and determine if downloading just to get etag (useStash) or to save file locally & get etag (!useStash)
+     * @param {AbstractSession} session - the session object generated from the connected profile
+     * @param {ILocalFile} lfFile - object containing pertinent information about the local file during the editing process
+     * @param {boolean} useStash - should be true if don't want to overwrite local file when refreshing etag
+     * @returns {ILocalFile}
+     */
+    public static async localDownload(session: AbstractSession, lfFile: ILocalFile, useStash: boolean): Promise<ILocalFile>{
+        // account for both useStash|!useStash and uss|ds when downloading
+        const tempPath = useStash ? path.posix.join(tmpdir(), "toDelete.txt") : lfFile.tempPath;
+        const args: [AbstractSession, string, IDownloadOptions] = [
+            session,
+            lfFile.fileName,
+            {
+                returnEtag: true,
+                file: tempPath
+            }
+        ];
+
+        if(lfFile.fileType === 'uss'){
+            lfFile.zosResp = await Download.ussFile(...args);
+        }else{
+            lfFile.zosResp = await Download.dataSet(...args);
+        }
+
+        if (useStash){
+            this.destroyTempFile(path.posix.join(tmpdir(), "toDelete.txt"));
+        }
+        return lfFile;
+    }
+
+    /**
+     * Performs appropriate file comparison (either in browser or as a terminal diff) between lf-USS or lf-DS.
+     * Local file (lf) will then be opened in default editor
+     * @param {AbstractSession} session - the session object generated from the connected profile
+     * @param {IHandlerParameters} commandParameters - parameters supplied by args
+     * @returns {Promise<IZosFilesResponse>} - the response generated by {@link CompareBaseHelper.getResponse}
+     * @memberof EditUtilities
+     */
+    public static async fileComparison(session: AbstractSession, commandParameters: IHandlerParameters): Promise<IZosFilesResponse>{
+        const handlerDs = new LocalfileDatasetHandler;
+        const handlerUss = new LocalfileUssHandler;
+        const helper = new CompareBaseHelper(commandParameters);
+        const gui = ProcessUtils.isGuiAvailable();
+        const options: IDiffOptions = {
```

Comment:
This is because a change was also made in imperative, you'll have to pull those recent changes too to test https://github.com/zowe/imperative/pull/967